examples/go-client/ruleEngine/engine.go

```diff
@@ -0,0 +1,127 @@
+// +build sample_client
+
+package ruleEngine
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/Comcast/rulio/core"
+	"github.com/Comcast/rulio/cron"
+	"github.com/Comcast/rulio/examples/go-client/configuration"
+	"github.com/Comcast/rulio/examples/go-client/service"
+	"github.com/Comcast/rulio/sys"
+	"os/exec"
+	"strings"
+	"time"
+)
+
+type EnginePoc struct {
+	Ctx     *core.Context
+	Service *service.Service
+}
+
+func NewEngine(envConfig *configuration.AppEnvConfigPoc, ctx *core.Context) (*EnginePoc, error) {
+
+	inMemoryConf := sys.ExampleConfig()
+	if envConfig.LinearState {
+		inMemoryConf.UnindexedState = true
+	}
+	inMemoryConf.Storage = envConfig.StorageType
+	inMemoryConf.StorageConfig = envConfig.StorageConfig
+
+	cont := sys.ExampleSystemControl()
+	cont.MaxLocations = envConfig.MaxLocations
+	ttl, err := sys.ParseLocationTTL(envConfig.LocationTTL)
+	if err != nil {
+		return nil, err
+	}
+	cont.LocationTTL = ttl
+
+	var cronner cron.Cronner
+	if envConfig.CronURL == "" {
+		cr, _ := cron.NewCron(nil, time.Second, "intcron", 1000000)
+		go cr.Start(ctx)
+		cronner = &cron.InternalCron{Cron: cr}
+	} else {
+		cronner = &cron.CroltSimple{
+			CroltURL: envConfig.CronURL,
+			RulesURL: envConfig.RulesURL,
+		}
+	}
+
+	verb, err := core.ParseVerbosity(envConfig.Verbosity)
+	if err != nil {
+		panic(err)
+	}
+	locCtl := &core.Control{
+		MaxFacts:  envConfig.MaxFacts,
+		Verbosity: verb,
+	}
+	if envConfig.BashActions {
+		locCtl.ActionInterpreters = map[string]core.ActionInterpreter{
+			"bash": &BashActionInterpreter{},
+		}
+	}
+	cont.DefaultLocControl = locCtl
+
+	// TODO: New system is being created
+	sys, err := sys.NewSystem(ctx, *inMemoryConf, *cont, cronner)
+	if err != nil {
+		return nil, err
+	}
+	ctx.SetLogValue("app.id", "rulesys")
+	core.UseCores(ctx, false)
+	ctx.Verbosity = verb
+	ctx.LogAccumulatorLevel = verb
+
+	engineService := &service.Service{
+		System:  sys,
+		Router:  nil,
+		Stopper: nil,
+	}
+	return &EnginePoc{Ctx: ctx, Service: engineService}, nil
+}
+
+type BashActionInterpreter struct {
```

Comment:
Probably shouldn't duplicate code like this. Can you just reference/use existing types instead of copy/pasting code?